From 1629e772018475f673f0dcd51b5ef80ca756e696 Mon Sep 17 00:00:00 2001
From: Motohiro Kosaki <mkosaki@redhat.com>
Date: Mon, 25 Aug 2014 22:51:33 -0400
Subject: [cpufreq] acpi-cpufreq: skip loading acpi_cpufreq after intel_pstate

Message-id: <1409007096-2821-4-git-send-email-mkosaki@redhat.com>
Patchwork-id: 89067
O-Subject: [RHEL7.1 BZ1116286 PATCH 3/6] acpi-cpufreq: skip loading acpi_cpufreq after intel_pstate
Bugzilla: 1123250
RH-Acked-by: Lenny Szubowicz <lszubowi@redhat.com>
RH-Acked-by: Larry Woodman <lwoodman@redhat.com>

From: KOSAKI Motohiro <mkosaki@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1123250
Tree: linus
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7878648
Changes-from-upstream: none

commit 8a61e12e84597b5f8155ac91b44dea866ccfaac2
Author: Yinghai Lu <yinghai@kernel.org>
Date:   Fri Sep 20 10:43:56 2013 -0700

    acpi-cpufreq: skip loading acpi_cpufreq after intel_pstate

    If the hw supports intel_pstate and acpi_cpufreq, intel_pstate will
    get loaded first.

    acpi_cpufreq_init() will call acpi_cpufreq_early_init()
    and that will allocate perf data and init those perf data in ACPI core,
    (that will cover all CPUs). But later it will free them as
    cpufreq_register_driver(acpi_cpufreq) will fail as intel_pstate is
    already registered

    Use cpufreq_get_current_driver() to check if we can skip the
    acpi_cpufreq loading.

    Signed-off-by: Yinghai Lu <yinghai@kernel.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: KOSAKI Motohiro <mkosaki@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/acpi-cpufreq.c b/drivers/cpufreq/acpi-cpufreq.c
index 748c6e1..a09e420 100644
--- a/drivers/cpufreq/acpi-cpufreq.c
+++ b/drivers/cpufreq/acpi-cpufreq.c
@@ -991,6 +991,10 @@ static int __init acpi_cpufreq_init(void)
 {
  int ret;
 
+ /* don't keep reloading if cpufreq_driver exists */
+ if (cpufreq_get_current_driver())
+  return 0;
+
  if (acpi_disabled)
   return 0;
 
-- 
1.7.1